rfcs/20240903-onemkl-interface-rename/README.md

```diff
@@ -0,0 +1,205 @@
+issue on multiple occasion internally when Codeplay started to contribute to
+oneMKL Interface. The oneMKL Interface
+[README](https://github.com/oneapi-src/oneMKL?tab=readme-ov-file#onemkl) already
+explains the differences between oneMKL Interface and Intel oneMKL product but
+this is not enough for people who are not working directly on the projects.
+
+## Outline
+
+1. [Introduction](#introduction)
+2. [Proposal](#proposal)
+3. [User impact](#user-impact)
+4. [Open questions](#open-questions)
+
+## Introduction
+
+oneMKL Interface has historically been implemented and supported by Intel. As
+the project is moving to the UXL foundation we want to avoid using the name of
+an Intel product. This RFC describes the solution of renaming oneMKL Interface
+to a new name.
+
+## Proposal
+
+The main purpose of this RFC is to agree on a new name for the oneMKL specification and oneMKL Interface.
+Some of the suggested names for the implementations are:
+* **oneMath**
+  * The specification would be renamed oneMath Specification
+  * The C++ namespace would be `oneapi::math`
+  * The CMake namespace would be `ONEMATH`
+* **oneSLA** (SYCL Linear Algebra)
+  * The specification would be renamed oneSLA Specification
+  * New namespace would be `oneapi::sla`
+  * The CMake namespace would be `ONESLA`
+
+Other suggestions are welcomed. The name **oneMath** will be chosen if there are
+no objections by October 4, 2024.
+
+The suggested solution is to proceed in the following steps:
+1. The UXL foundation agrees on the new name.
+2. Codeplay submits a oneAPI-spec PR to rename the occurrences of "oneMKL" and
+   "mkl" to the new name.
+3. Codeplay submits a oneMKL Interface PR to:
+  * Update the root README to use the new name, with a mention that the project
+    was formerly called oneMKL Interface.
+  * Update the references to "oneMKL" and `onemkl_` in the documentation as
+    seen in the first few lines of
+    [docs/onemkl-datatypes.rst](https://github.com/oneapi-src/oneMKL/blob/develop/docs/onemkl-datatypes.rst?plain=1#L1)
+    for instance.
+  * Update occurrences of "onemkl" in internal functions such as
+    [onemkl_cublas_host_task](https://github.com/oneapi-src/oneMKL/blob/1ce98a699f93bd3a78350269b2e34d822fe43b91/src/blas/backends/cublas/cublas_task.hpp#L77).
+  * Update macros such as include guards and other internal macros like
+    `ONEMKL_EXPORT` to use the new name.
+  * Rename CMake targets `onemkl` and `onemkl_<domain>_<backend>` to use the
+    new name. The existing targets name can be added with a deprecation
+    messages for anyone using them. See the section on [CMake target
+    deprecation](#cmake-deprecated-target) for more details. The namespace of
+    the exported and installed targets are changed to use the new name.
+  * The namespace `oneapi::mkl` is deprecated and aliased to the new namespace
+     name.
+  * The main header and domain headers are deprecated and include the new headers as shown below:
```

Comment:
I think this will run into this issue with the rng domain: https://github.com/oneapi-src/oneMKL/issues/438
I assumed `rng/device.hpp` should be included by `rng.hpp` but I can see it is explicitly mentioned in the [README](https://github.com/oneapi-src/oneMKL?tab=readme-ov-file#device-api) that the file should be included separately. Can someone from @oneapi-src/onemkl-rng-write confirm that this is intentional? If so we should make sure that `include/oneapi/mkl/rng/device.hpp` is deprecated and includes the new equivalent header.

Reply:
@Rbiessy, yes, having a standalone rng/device.hpp is intentional, as we wanted to stress out what device API are a completely different entity and has header-based implementation. So the expectation is that users may independently include rng/device.hpp functionality within sycl kernels and doesn't need to link to any binaries.